_observing-your-data/ad/index.md

```diff
@@ -32,88 +32,113 @@ A _detector_ is an individual anomaly detection task. You can define multiple de
+
+If the custom index you specify does not exist, the Anomaly Detection plugin will create it when you create the detector and start your real-time or historical analysis.
+
+If the custom index already exists, the plugin will verify that the index mapping matches the required structure for anomaly results. In this case, ensure that the custom index has a valid mapping as defined in the [`anomaly-results.json`](https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/resources/mappings/anomaly-results.json) file.
+To use the custom results index option, you must have the following permissions:To use the custom results index option, you must have the following permissions:
+
+- `indices:admin/create` -- The `create` permission is required in order to create and roll over the custom index.
+- `indices:admin/aliases` -- The `aliases` permission is required in order to create and manage an alias for the custom index.
+- `indices:data/write/index` -- The `write` permission is required in order to write results into the custom index for a single-entity detector.
+- `indices:data/read/search` -- The `search` permission is required in order to search custom results indexes to show results on the Anomaly Detection interface.
+- `indices:data/write/delete` -- The detector may generate many anomaly results. The `delete` permission is required in order to delete old data and save disk space.
+- `indices:data/write/bulk*` -- The `bulk*` permission is required because the plugin uses the Bulk API to write results into the custom index.
+
+##### Flattening nested fields
+
+Custom result index mappings with nested fields pose aggregation and visualization challenges. The **Enable flattened custom result index** flattens the nested fields in the custom result index. When selecting this option, the plugin creates a separate index prefixed with the custom result index name and detector name. 
+
+For example, if the detector `Test` uses the custom result index `abc`, a separate index with the alias `opensearch-ad-plugin-result-abc-flattened-test` will store the anomaly detection results with nested fields flattened. Besides creating a separate index, the plugin also sets up an ingest pipeline with a script processor. This pipeline is bound to the separate index and uses a Painless script to flatten all nested fields in the custom results index.
+
+Deactivating this option on a running detector removes its flattening ingest pipeline; it also ceases to be the result index’s default.
+When using the flattened custom result inside option, consider the following
+
+- Anamoly Detection constructs the index name based on the custom result index and detector name, and since the detector name is editable, conflicts can occur. If a conflict occurs, anamoly detection reuses the index name
```

Comment:
```suggestion
- Anomaly Detection constructs the index name based on the custom result index and detector name, and since the detector name is editable, conflicts can occur. If a conflict occurs, anomaly detection reuses the index name
```